File: javascript/resistor-color-duo/resistor-color-duo.js
export const decodedValue = (resistorColors) => {
  return colorCode(resistorColors[0]) * 10 + colorCode(resistorColors[1]);
};

const colorCode = (color) => {
  return COLORS.indexOf(color);
};

const COLORS = [
  "black",
  "brown",
  "red",
  "orange",
  "yellow",
  "green",
  "blue",
  "violet",
  "grey",
  "white"
];

File: ruby/resistor-color-duo/resistor_color_duo.rb
class ResistorColorDuo
  RESISTOR_VALUES = {
    black: 0,
    brown: 1,
    red: 2,
    orange: 3,
    yellow: 4,
    green: 5,
    blue: 6,
    violet: 7,
    grey: 8,
    white: 9
  }
  
  def self.value(colors)
    RESISTOR_VALUES[colors[0].to_sym] * 10 + RESISTOR_VALUES[colors[1].to_sym] 
  end
end